From b6a62a5558490d727b2c4aa4fc964dcde1510ae8 Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Fri, 11 Nov 2022 09:32:20 +0800
Subject: [PATCH] mm: reliable: Return corrent errno in reliable_check
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b6a62a5558490d727b2c4aa4fc964dcde1510ae8
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SK3S
CVE: NA

--------------------------------

LTP's proc01 test was failing be because this ret code (1):

	proc01      1  TFAIL  :  proc01.c:400: read failed: /proc/self/task/1406366/reliable: errno=EPERM(1): Operation not permitted

To slove this problem, return corrent errno in reliable_check().

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/proc/base.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/proc/base.c b/fs/proc/base.c
index 9b4666e757f0..cc1fdff2e136 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -1264,14 +1264,14 @@ static const struct file_operations proc_oom_score_adj_operations = {
 static inline int reliable_check(struct task_struct *task, struct pid *pid)
 {
 	if (!mem_reliable_is_enabled())
-		return -EPERM;
+		return -EACCES;
 
 	if (is_global_init(task))
-		return -EPERM;
+		return -EINVAL;
 
 	if (!task->mm || (task->flags & PF_KTHREAD) ||
 	    (task->flags & PF_EXITING))
-		return -EPERM;
+		return -EINVAL;
 
 	return 0;
 }
-- 
2.33.0

